### ISSUE TITLE
When one postpones the reminder into the next day the yes it's counted for the next day.

### ISSUE THREAD

**Issue by vonneudeck** [iSoron/uhabits#1174]
## Pre-submission checklist
- [x] I am submitting a bug report, not a feature request.
- [x] this is related to #969 

## Description
When one postpones the reminder into the next day the yes is counted for the next day.
While technically correct, this is socially wrong because when I take my pills at 00:20 on Monday, they should still count to my evening pill taking habit for Sunday.

## Steps to reproduce
1. Create a habit with a reminder at 22:00
1. Postpone the reminder twice for an hour by pressing "later" in notification
1. Press "yes" in the notification after midnight
1. Enjoy having the habit fulfilled for the next day instead of the original day.

## System information
 - Phone: Sony Xperia H8314
 - Phone Operating System: Android 10
 - App version: 2.0.3

**Comment by hiqua** [comment]
> this is socially wrong because when I take my pills at 00:20 on Monday, they should still count to my evening pill taking habit for Sunday.

I guess this is debatable, but the one thing we should do if we don't do it yet is to change this behavior if the "extend day a few hours past midnight" option is active.

Are you saying it's not the case? With this option, does checking the reminder apply to the previous or to the current day?

**Comment by vonneudeck** [comment]
I did not activate the "extend the day a few hours option", but I activated it now. I'll try it next time I am awake after midnight. I understand why it was technically more easy to implement short term. But I don't think that option should play a role for this case.
If my reminder was set for the last day 22:00 , and I fulfil my habit by pressing the reminder after postponing it beyond midnight the app should ideally have a state with a date when the reminder was fired the first time and use that.

But I understand that maintaining software is a lot of work, so if you decide not to change that, I am of course very fine with that.
